File: tests/run_model.R

library(gelnet)

A <- as.matrix(read.csv("train.matrix", sep="\t", row.names=1))
y <- as.matrix(read.csv("train.labels", sep="\t", row.names=1))
X <- as.matrix(read.csv("train.features", sep="\t", row.names=1))

L <- adj2nlapl(A)

gelnet( X, y, 0.1, 1, P = L, max.iter=100 )

File: tests/run_model.py
#!/usr/bin/env python

import sys
import pandas
import py_gelnet
import numpy as np

with open(sys.argv[1] + ".matrix") as handle:
    A = pandas.read_csv(handle, sep="\t", index_col=0)
L = py_gelnet.adj2nlapl(A.as_matrix())

with open(sys.argv[1] + ".features") as handle:
    X = pandas.read_csv(handle, sep="\t", index_col=0)

with open(sys.argv[1] + ".labels") as handle:
    y = pandas.read_csv(handle, sep="\t", index_col=0)

X_in = X.as_matrix().copy(order="fortran")
y_in = y.as_matrix().reshape(y.shape[0])

model = py_gelnet.gelnet( X_in, y_in, 0.1, 1, P = L, max_iter=100 )
print model
